specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json

```diff
@@ -3157,10 +3164,47 @@
           "items": {
             "type": "string"
           }
+        },
+        "identityInfo": {
+          "$ref": "#/definitions/IdentityInfo",
+          "description": "Managed Identity information required to access customer storage account."
         }
       },
       "x-ms-discriminator-value": "IaasVMRestoreRequest"
     },
+    "IdentityDetails": {
+      "type": "object",
+      "properties": {
+        "isSystemAssignedIdentity": {
+          "type": "boolean"
+        },
+        "managedIdentityArmUrl": {
+          "type": "string"
+        }
+      }
+    },
+    "IdentityInfo": {
+      "description": "Encapsulates Managed Identity related information",
+      "type": "object",
+      "properties": {
+        "objectType": {
+          "description": "Type of the specific object - used for deserializing",
+          "type": "string"
+        },
+        "isSystemAssignedIdentity": {
```

Comment:
> "isSystemAssignedIdentity": { [](http://example.com/codeflow?start=7&length=30)

Having a bool flag is not recommended. Please use a string enum type.

Reply:
In past ARM PR review we got this suggestion from ARM team that, having a “Type” and maintaining an enum is an overhead and leads to issues. With this reasoning we were suggested to use bool for a similar property.
We already use this in existing API. For this, we prefer to keep it as is.